Clean up syntax documentation

GiudGiud opened the pull request:
Fixing:

Bounds & auxkernels appear within the same syntax when they should not (not inter-operable in any way, they share the same base class but that's it)
And controllogic and Controls being mixed (same concern)
And FunctorMateirals and Materials (same concern, we have been doing fine mixing them in NS but it s not ideal. If a FE implementation of NS starts using functor materials (with materials already available) it will be big time confusing)

~~I have not figured out how to prevent all variables to appear as options under markers. All variables clearly cannot be used as Markers (need states, not Real values for that) unless one is hacking badly.~~

I added a deprecation option because otherwise we ll get some complaints, FunctorMaterials in the [Materials] block is still very common

EDIT:
I ended up finding the unregister.yml that addresses the syntax concerns much easier. The syntax page is listing much less invalid syntax.
However fixing the documentation without fixing the code is not a good idea, everything else I did is still necessary.

GiudGiud committed:
Separate functormaterials and materials in input
refs #25466

GiudGiud committed:
Move NS module functormaterials to functor materials folder
refs #25466

GiudGiud committed:
Adapt NS tests to new name
refs #25466

GiudGiud committed:
Enable deprecated object types for tasks registered to only being allowed to add certain objects
This will smooth the transition for functormaterials/materials
refs #25466

GiudGiud committed:
Separate control logic (THM controls) and controls
- they are not that interoperable
refs #25466

GiudGiud committed:
Deprecate ability to write bounds in auxkernels
clearer to use the dedicated syntax
refs #25466

GiudGiud committed:
Move bounds out of auxkernels
They are not auxkernels, they dont set an aux-variable
refs #25466

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1956528/) on 6fcd6ed wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/26377/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1961005/) on 37dfdc8 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/26377/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/26377/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 0ae736620977bf6741d683bcebb80582733d77e7`

GiudGiud committed:
Achieve everything I wanted to do the easy way:
- unregister objects from syntaxes they dont belong to
Fixes variables showing up in Markers syntax

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1963882/) on 1628faa wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>837979</th>
      <th colspan=4>#26377 1628fa</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.07%</td>
      <td>85.07%</td>
      <td>+0.00%</td>
      <td>94.94%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>98357</td>
      <td>98374</td>
      <td>+17</td>
      <td>75</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>17264</td>
      <td>17263</td>
      <td>-1</td>
      <td>4</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/26377/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/26377/coverage/framework)


## Modules coverage

### Navier stokes

<table>
  <thead>
    <tr>
      <th></th>
      <th>837979</th>
      <th colspan=4>#26377 1628fa</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>84.22%</td>
      <td>84.22%</td>
      <td>+0.00%</td>
      <td>93.30%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>13194</td>
      <td>13195</td>
      <td>+1</td>
      <td>195</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>2473</td>
      <td>2473</td>
      <td>-</td>
      <td>14</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/26377/coverage_diff/navier_stokes)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/26377/coverage/navier_stokes)

### Thermal hydraulics

<table>
  <thead>
    <tr>
      <th></th>
      <th>837979</th>
      <th colspan=4>#26377 1628fa</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>89.07%</td>
      <td>89.08%</td>
      <td>+0.00%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>13866</td>
      <td>13870</td>
      <td>+4</td>
      <td>5</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1701</td>
      <td>1701</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/26377/coverage_diff/thermal_hydraulics)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/26377/coverage/thermal_hydraulics)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/26377/coverage/xfem)

</details>



This comment will be updated on new commits.

GiudGiud commented:
I ll probably patch the apps doco after Jan 9 but otherwise ready.

GiudGiud committed:
Rename all bounds objects to remove the aux suffix

GiudGiud committed:
Adapt modules to new name for Bounds

GiudGiud committed:
Rebase and move kEps functor material

lindsayad commented on framework/doc/content/source/bounds/VariableOldValueBounds.md:
```diff
@@ -0,0 +1,23 @@
+`variable`.
+
+Note that in order for these bounds to have an effect, the user has to specify the
+PETSc options `-snes_type vinewtonssls` or `-snes_type vinewtonrsls`. A warning will be generated if neither options are specified. MOOSE users can also specify bound with a constant using [`ConstantBounds`](/ConstantBounds.md).
```
```suggestion
PETSc options `-snes_type vinewtonssls` or `-snes_type vinewtonrsls`. A warning will be generated if neither option is specified. MOOSE users can also specify bound with a constant using [`ConstantBounds`](/ConstantBounds.md).
```

lindsayad commented on framework/include/parser/Syntax.h:
```diff
@@ -207,6 +211,10 @@ class Syntax
   /// The list of Moose system objects to tasks.  This map indicates which tasks are allowed to build certain MooseObjects.
   std::multimap<std::string, std::string> _moose_systems_to_tasks;
 
+  /// The list of deprecated Moose system objects to tasks. If an item is in this map, the task is still allowed to build
+  /// the object from the MOOSE system, but it is deprecated to do so.
+  std::multimap<std::string, std::string> _deprecated_moose_systems_to_tasks;
```
It would be good to try and clarify this member more. The first sentence of the doxygen and the name of the member suggest that the system itself is deprecated. But I believe you are only using this for adding `FuctorMaterials` under `add_material`; `FunctorMaterials` themselves are not a deprecated system, but adding them under `Materials` syntax with the `add_material` task is (if I understand correctly)

GiudGiud replied:
yeah it s deprecated ( moose systems to tasks ).
instead of moose_depcreated_systems_to_tasks.

GiudGiud replied:
renamed and change docstring

lindsayad commented on framework/src/parser/Syntax.C:
```diff
@@ -314,6 +319,15 @@ Syntax::verifyMooseObjectTask(const std::string & base, const std::string & task
+  for (const auto & task_it : as_range(iters))
+    if (task == task_it.second)
+    {
+      mooseDeprecated("Adding '" + base + "' though task '" + task +
```
```suggestion
      mooseDeprecated("Adding '" + base + "' through task '" + task +
```

lindsayad commented on framework/src/parser/Syntax.C:
```diff
@@ -314,6 +319,15 @@ Syntax::verifyMooseObjectTask(const std::string & base, const std::string & task
+    if (task == task_it.second)
+    {
+      mooseDeprecated("Adding '" + base + "' though task '" + task +
+                      "' is deprecated. The syntax you are using is deprecated.");
```
Second sentence seems redundant

GiudGiud replied:
changed

GiudGiud replied:
ok I made it as obvious as possible what is changing / happening in the deprecation message. Maybe even too obvious

lindsayad commented on modules/navier_stokes/doc/content/source/functormaterials/INSFVkEpsilonViscosityMaterial.md:
```diff
@@ -13,8 +13,8 @@ where:
-!syntax parameters /Materials/INSFVkEpsilonViscosityMaterial
+!syntax parameters /FunctorMaterials/INSFVkEpsilonViscosityMaterial
 
-!syntax inputs /Materials/INSFVkEpsilonViscosityMaterial
+!syntax inputs /FunctorMaterials/INSFVkEpsilonViscosityMaterial
 
-!syntax children /Materials/INSFVkEpsilonViscosityMaterial
+!syntax children /FunctorMaterials/INSFVkEpsilonViscosityMaterial
```
To be consistent with your other changes seems like this should now have the suffix `FunctorMaterial`

GiudGiud replied:
will do

GiudGiud replied:
done

lindsayad commented on modules/thermal_hydraulics/src/parser/THMSyntax.C:
```diff
@@ -24,7 +24,8 @@ associateSyntax(Syntax & syntax)
   syntax.registerActionSyntax("AddComponentAction", "Components/*/*", "THM:add_component");
   syntax.registerActionSyntax("AddClosuresAction", "Closures/*", "THM:add_closures");
-  syntax.registerActionSyntax("THMAddControlAction", "ControlLogic/*", "add_control");
+  syntax.registerActionSyntax("THMAddControlAction", "ControlLogic/*", "THM:add_control");
+  syntax.registerTaskName("THM:add_control", "THMControl", false);
```
Would it be better to to change `THM::add_control` to `add_control_logic`? @joshuahansel

joshuahansel replied:
It's a little messy. The base class is `THMControl` even though the syntax/system is `ControlLogic`. The other tasks we've registered in THM are preceded with `THM::`. `THM::add_control_logic` is probably best, but I think either way is fine.

GiudGiud replied:
changed

GiudGiud committed:
Address Alex's review
- clarify depcreation message
- clarify docstring for new list of systems to tasks

Co-authored-by: Alex Lindsay <alexander.lindsay@inl.gov>

Update framework/src/parser/Syntax.C

Co-authored-by: Alex Lindsay <alexander.lindsay@inl.gov>

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1982600/) on dc6f560 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/26377/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/26377/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 28bf8f4f287b56d948c110e55ea724380f959b56`

lindsayad reviewed (approved):
failures are unrelated

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1984549/) on b96b592 wanted to post the following:


The following coverage requirement(s) failed:
- Failed to generate `framework` coverage rate (required: 81.0%)
